Update Row-level TTL cron job durations, etc.

rmloveland opened the pull request:
Fixes [DOC-9338](https://cockroachlabs.atlassian.net/browse/DOC-9338) by updating ~all TTL examples to set the `ttl_job_cron` value to `@daily`, since the default (`@hourly`) is really too short.

Fixes [DOC-8836](https://cockroachlabs.atlassian.net/browse/DOC-8836) (for v23.2)

Touches the obvious/easy parts of [DOC-9443](https://cockroachlabs.atlassian.net/browse/DOC-9443) (TIMESTAMP -> TIMESTAMPTZ)

NB. A subset of the v23.2 changes were also ported to v22.2 and v23.1

annrpom commented:
lgtm so far! Before this merges, I believe we also want to change the default here to `@daily` (do we usually do a `new in 23.2` callout for stuff like this?)
https://github.com/cockroachdb/docs/blob/eb506826c512e9c47b96173523dd9e1aa40231a8/src/current/v23.2/row-level-ttl.md?plain=1#L148

rmloveland commented:
> lgtm so far! Before this merges, I believe we also want to change the default here to `@daily` (do we usually do a `new in 23.2` callout for stuff like this?)
> 
> https://github.com/cockroachdb/docs/blob/eb506826c512e9c47b96173523dd9e1aa40231a8/src/current/v23.2/row-level-ttl.md?plain=1#L148

Updated to say `@daily` and added the "new in v23.2" callout thingy :-) 

Now it says

> Default: `'@daily'` (was `'@hourly'`)

PS separately, if this change is LGTM'd I plan to backport it to the v22.2 and v23.1 docs but will remove the bit about "@daily" being the default because it isn't yet in those versions (at least I did not find a backport of https://github.com/cockroachdb/cockroach/pull/110623, please correct me if there is one)

annrpom reviewed (approved):
awesome; thanks - lgtm!
i sanity checked that the change was not backported by `git tag --contains 0b5a0a6a2c69cd68fd6db86358818a3664093855` in the cockroach repo

taroface reviewed (approved):
LGTM!
